packages/utils/src/types/primitives.ts

```diff
@@ -120,10 +120,18 @@ export class Address {
 }
 
 export class Hash {
-  constructor(private arr: Uint8Array) {
+  protected arr: Uint8Array
+  constructor(arr: Uint8Array) {
     if (arr.length !== Hash.SIZE) {
       throw new Error('Incorrect size Uint8Array for hash')
     }
+
+    if (typeof Buffer != undefined && Buffer.isBuffer(arr)) {
```

Comment:
Why doesn't typescript prevent buffers being used here?

Reply:
See your comment above :P

Reply:
in that case, why not throw if it's a buffer so we can actually fix these errors.

Reply:
done, led to various other fixes that were detected after throwing an error